src/lib/sparse-buffer.ts

```diff
@@ -319,7 +325,38 @@ export class SparseBuffer implements ISparseBuffer {
             undefined
         );
     }
+}
+
+function trim(result: INumberArray) {
+    result.length = result.count;
+    (result as any).count = undefined;
+}
 
+export function shuffle(numbers: number[]): void {
```

Comment:
Just so I understand, are we shuffling ids or segments? Because segments have ordered ids, if we are only ordering segments, I'm worried this will localise the random selection - especially because the campaigns that will use this will be choosing very small customer sets (around 100-300 customers).

Reply:
Segments would be buffers -- not numbers. This shuffling happens after the numbers are unpacked out of sparse buffers.